include/swift/AST/ClangModuleLoader.h

```diff
@@ -58,6 +60,30 @@ class ClangModuleLoader : public ModuleLoader {
                                             const DeclContext *overlayDC,
                                             const DeclContext *importedDC) = 0;
 
+  /// Look for declarations associated with the given name.
```

Comment:
These virtual methods are necessary to break a cyclic dependency on Linux.

Reply:
Can you explain further? I'm not sure why there'd be a cycle.

Reply:
The problem only existed on Linux, and no amount of duplicating llink libraries in the CMakeLists would make it go away. Part of the problem was that ASTDemangler in libAST.a had an explicit cast and a method call to DWARFImporter and libDWARFImporter.a in turn depends on libAST. I do not understand why this wasn't also a problem with ClangImporter, which effectively does the same thing. If you are concerned about the interface/performance I can try a little harder, but I also think having a common interface in the common base class of ClangImporter and DWARFImporter is actually good.

Reply:
ASTDemangler definitely should not be calling DWARFImporter directly, and I'm sorry I missed that part. Let's come up with a better way to let DWARFImporter intervene in looking up Clang types.